websites/G/Geotastic/presence.ts

```diff
@@ -0,0 +1,142 @@
+const presence = new Presence({
+		clientId: "978557181911773214"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
+		details: "Viewing an unsupported page",
+		startTimestamp: browsingTimestamp
+	};
+
+	switch (document.location.pathname) {
+		case "/home": {
+			presenceData.details = "Viewing the home page";
+			break;
+		}
+		case "/local-lobby": {
+			presenceData.details = "Creating a local lobby";
+			break;
+		}
+		case "/user-challenges/find": {
+			presenceData.details = "Browsing unplayed user challenges";
+			break;
+		}
+		case "/user-challenges/played": {
+			presenceData.details = "Browsing played user challenges";
+			break;
+		}
+		case "/user-challenges/own": {
+			presenceData.details = "Browing created user challenges";
+			break;
+		}
+		case "/user-challenges/create": {
+			presenceData.details = "Creating a challenge";
+			break;
+		}
+		case "/matchmaking-lobby": {
+			presenceData.details = "In a matchmaking lobby";
+			break;
+		}
+		case "/community-map": {
+			presenceData.details = "Viewing the community map";
+			break;
+		}
+		case "/squad/list": {
+			presenceData.details = "Browsing the list of squads";
+			break;
+		}
+		case "/squad/management": {
+			presenceData.details = "Viewing the squad management page";
+			break;
+		}
+		case "/map-editor/maps": {
+			presenceData.details = "Creating a map";
+			break;
+		}
+		case "/map-editor/drop-groups": {
+			presenceData.details = "Viewing their drop groups";
+			break;
+		}
+		case "/events": {
+			presenceData.details = "Browsing past and future events";
+			break;
+		}
+		case "/account/profile": {
+			presenceData.details = "Viewing their profile settings";
+			break;
+		}
+		case "/account/statistics": {
+			presenceData.details = "Viewing their statistics";
+			break;
+		}
+		case "/account/trophies": {
+			presenceData.details = "Viewing their trophies";
+			break;
+		}
+		case "/account/friends": {
+			presenceData.details = "Viewing their friends";
+			break;
+		}
+		case "/account/supporter-status": {
+			presenceData.details = "Viewing their supporter status";
+			break;
+		}
+		case "/account/quota-consumption": {
+			presenceData.details = "Viewing their quota consumption";
+			break;
+		}
+		case "/account/account-settings": {
+			presenceData.details = "Viewing their account settings";
+			break;
+		}
```

Comment:
Please use a Record for this instead as shown below. Much shorter and more maintainable 
https://github.com/PreMiD/Presences/blob/113adabd2296ec1be22122ccc3ec2deff1c399ed/websites/T/TruckersMP/presence.ts#L15-L41